src/rospkg/manifest.py

```diff
@@ -370,14 +370,17 @@ def _get_text(nodes):
 _static_rosdep_view = None
 
 
-def parse_manifest_file(dirpath, manifest_name, rospack=None):
+def parse_manifest_file(dirpath, manifest_name, rospack=None, typeof_depend=0):
     """
     Parse manifest file (package, stack).  Type will be inferred from manifest_name.
 
     :param dirpath: directory of manifest file, ``str``
     :param manifest_name: ``MANIFEST_FILE`` or ``STACK_FILE``, ``str``
     :param rospack: a RosPack instance to identify local packages as ROS packages
-
+    :param typeof_depend:
+            0: all kinds of depend.
+            1: only runtime_depend, such as run_depend in package.xml format 1 and 2.
+            2: only build_depend.
```

Comment:
Please do not use an opaque numeric value for this purpose. Instead pass an iterable with the names of the dependency types (similar to other ROS API, e.g. https://github.com/ros-infrastructure/rosdistro/blob/26c9dbfc9634b0dd1797509f981477f51b736839/src/rosdistro/dependency_walker.py#L66).